test/retry_test.go

```diff
@@ -579,7 +581,7 @@ func (s) TestRetryStats(t *testing.T) {
 	// There is a race between receiving the payload (triggered by the
 	// application / gRPC library) and receiving the trailer (triggered at the
 	// transport layer).  Adjust the received stats accordingly if necessary.
-	const tIdx, pIdx = 13, 14
+	const tIdx, pIdx = 15, 16
```

Comment:
Existing tests should probably filter out `PickerUpdated` events entirely, unless we know they are deterministic in those tests.  And we should have a new test with a custom LB policy that confirms those events work as intended.

Reply:
Done.